CADETProcess/modelBuilder/ZRMBuilder.py

```diff
@@ -0,0 +1,290 @@
+            segments_area: List of membrane segment areas m2.
+            binding_model: Optional binding model to apply to all zones.
+            name: Optional name for the builder instance.
+        """
+        # Initialize an empty list of zones/cstrs in and out;
+        # actual membrane zones will be appended later under specific identifiers
+        self.zones = []
+        self.cstrs_in = []
+        self.cstrs_out = []
+
+        self.component_system = component_system
+        self.flow_rate = flow_rate
+
+        # Checking negative values for flow rate
+        self.segments_area = segments_area
+        self.init_c = init_c
+        self.configuration = configuration.lower()
+        self._validate_inputs()
+
+        self.n = len(segments_area)  # Referring to the number of zones
+        self.binding_model = binding_model
+        self.name = name or 'ZRMBuilder'  # ZRMBuilder is the default name
+
+        self._flow_sheet = FlowSheet(component_system, self.name)
+        self._process = Process(self._flow_sheet, self.name)
+
+        # Define the inlet/outlet, with filling their attributes
+        self.inlet = Inlet(component_system, name='inlet')
+        self.inlet.c = self.init_c
+        self.inlet.flow_rate = self.flow_rate
+
+        self.outlet = Outlet(component_system, name='outlet')
+
+    @property
+    def flow_sheet(self) -> FlowSheet:
+        """FlowSheet: Return the FlowSheet instance associated with this builder."""
+        return self._flow_sheet
+
+    @property
+    def process(self)  -> Process:
+        """Process: Return the Process instance associated with this builder."""
+        return self._process
+
+    @property
+    def cycle_time(self)  -> float:
+        """float: Get the cycle time of the process."""
+        return self.process.cycle_time
+
+    @cycle_time.setter
+    def cycle_time(self, time: float)  -> None:
+        self._cycle_time = time
+
+    @property
+    def binding_model(self) -> None:
+        """
+        Get the binding model associated with this builder.
+
+        Parameters
+        ----------
+        binding_model : BindingBaseClass
+            The binding model to be retrieved.
+
+        Returns
+        -------
+        BindingBaseClass
+            The current binding model.
+        """
+        return self._binding_model
+
+    @binding_model.setter
+    def binding_model(self, binding_model: Optional['BindingBaseClass']) -> None:
+        """
+        Set the binding model for the zones in this builder.
+
+        Parameters
+        ----------
+        binding_model : BindingBaseClass
+            The binding model to be set for the zones.
+        """
+        self._binding_model = binding_model
+        for membrane_segment in self.zones:
+            membrane_segment.binding_model = binding_model
+
+    # returning the ratio of each zone membrane area from the total membrane area
+    def Segments_ratio(self) -> list[float]:
```

Comment:
Upper case variables are usually reserved for class definitions.

```suggestion
    def segments_ratio(self) -> list[float]:
```